pkg/clustermesh/service_merger.go

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package clustermesh
+
+import (
+	"github.com/cilium/hive/cell"
+
+	cmtypes "github.com/cilium/cilium/pkg/clustermesh/types"
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/loadbalancer/experimental"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/option"
+	serviceStore "github.com/cilium/cilium/pkg/service/store"
+	"github.com/cilium/cilium/pkg/source"
+)
+
+// ServiceMerger is the interface to be implemented by the owner of local
+// services. The functions have to merge service updates and deletions with
+// local services to provide a shared view.
+type ServiceMerger interface {
+	MergeExternalServiceUpdate(service *serviceStore.ClusterService, swg *lock.StoppableWaitGroup)
+	MergeExternalServiceDelete(service *serviceStore.ClusterService, swg *lock.StoppableWaitGroup)
+}
+
+type serviceMergerParams struct {
+	cell.In
+
+	ServiceCache k8s.ServiceCache
+	ExpConfig    experimental.Config
+	Writer       *experimental.Writer
+}
+
+func newServiceMerger(p serviceMergerParams) ServiceMerger {
+	if !p.ExpConfig.EnableExperimentalLB {
+		return p.ServiceCache
+	}
+	return &expServiceMerger{p.Writer}
+}
+
+type expServiceMerger struct {
+	writer *experimental.Writer
+}
+
+// MergeExternalServiceDelete implements k8s.ServiceCache.
+func (sm *expServiceMerger) MergeExternalServiceDelete(service *serviceStore.ClusterService, swg *lock.StoppableWaitGroup) {
+	if service.Cluster == option.Config.ClusterName {
```

Comment:
Nit. You can retrieve `cmtypes.ClusterInfo` rather than accessing the global variable.

Reply:
Ah great!